### PR TITLE
Better paths query

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes
- Reduce path query from querying each event order to a single query. improves runtime and allows for easy extendability for querying more depth
- make sure loader is shown when paths date filter is changed

## Checklist
- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)

### CODE REVIEW

**Comment by EDsCODE** [comment]
*some misc commit messages that creeped over

File: posthog/api/paths.py [hunk 1]
```diff
@@ -16,60 +17,61 @@ def _event_subquery(self, event: str, key: str):
...
         date_query = request_to_date_query(request.GET)
-        aggregate: QuerySet[PersonDistinctId] = PersonDistinctId.objects.filter(team=team)
 
-        aggregate = self._add_event_and_url_at_position(aggregate, team, 1, date_query)
-        urls: List[str] = []
+        sessions = Event.objects.filter(
+                team=team,
+                event='$pageview',
+                **date_query,
+                **{'properties__$current_url__isnull': False},
```
Comment (timgl) [inline posthog/api/paths.py:29]: I shaved about 2 seconds of 11 second runtime by removing this. I _think_ that's safe as every pageview really should have a url

**Comment by timgl** [comment]
The results seem slightly different on master vs your branch

Master
image

better-paths-query
image

Any idea why?

**Comment by EDsCODE** [comment]
One factor is that the event relations are determined within session windows so they will be shifted around a bit and I'd expect numbers to reduce overall. ~~But it's troubling when certain relations are completely different.~~ Those cutoffs also will be changing the path sequences. If you don't see anything egregiously wrong with the logic I can go back and comb through both master and my solution to figure out all the differences that might arise

**Comment by EDsCODE** [comment]
Going to add a test for paths that tests correctness. Events that occur after a previous event after 30 minutes (our defined session time) will be counted as a new root rather than the next event in a sequence

**Comment by timgl** [comment]
Hm okay I think a 30 minute cut off is fine for pageviews, but doesn't make too much sense to me for your other PR. Maybe we can revisit later though, happy with this for now.

**Review by timgl — APPROVED** [review]

**Comment by timgl** [comment]
(tests are failing though)

**Pull request merged**